homeassistant/components/crownstone/listeners.py

```diff
@@ -0,0 +1,145 @@
+"""
+Listeners for updating data in the Crownstone integration.
+
+For data updates, Cloud Push is used in form of an SSE server that sends out events.
+For fast device switching Local Push is used in form of UART USB with Bluetooth.
+"""
+from __future__ import annotations
+
+from functools import partial
+from typing import TYPE_CHECKING, cast
+
+from crownstone_core.packets.serviceDataParsers.containers.AdvExternalCrownstoneState import (
+    AdvExternalCrownstoneState,
+)
+from crownstone_core.packets.serviceDataParsers.containers.elements.AdvTypes import (
+    AdvType,
+)
+from crownstone_core.protocol.SwitchState import SwitchState
+from crownstone_sse.const import (
+    EVENT_ABILITY_CHANGE,
+    EVENT_ABILITY_CHANGE_DIMMING,
+    EVENT_SWITCH_STATE_UPDATE,
+)
+from crownstone_sse.events import AbilityChangeEvent, SwitchStateUpdateEvent
+from crownstone_uart import UartEventBus, UartTopics
+from crownstone_uart.topics.SystemTopics import SystemTopics
+
+from homeassistant.core import Event, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send, dispatcher_send
+
+from .const import (
+    DOMAIN,
+    SIG_CROWNSTONE_STATE_UPDATE,
+    SIG_UART_STATE_CHANGE,
+    SSE_LISTENERS,
+    UART_LISTENERS,
+)
+
+if TYPE_CHECKING:
+    from .entry_manager import CrownstoneEntryManager
+
+
+@callback
+def update_crwn_state_sse(manager: CrownstoneEntryManager, ha_event: Event) -> None:
```

Comment:
```suggestion
def async_update_crwn_state_sse(manager: CrownstoneEntryManager, ha_event: Event) -> None:
```

If its decorated with `@callback` we usually prefix it with `async_` to indicate its going to run in the event loop from the callee